src/megaclient.cpp

```diff
@@ -1340,9 +1375,31 @@ void MegaClient::exec()
                                 fetchnodes(true);
                                 reqtag = creqtag;
                             }
+                            else if (e == API_EAGAIN || e == API_ERATELIMIT)
+                            {
+                                if (!statecurrent)
+                                {
+                                    fnstats.eagains++;
+                                }
+                            }
                         }
                         // fall through
                     case REQ_FAILURE:
+                        if (pendingsc && !statecurrent)
```

Comment:
I agree. I put it that way because it might be required in the future to process more things while the filesystem is not current, but it's unlikely to have to process more things depending on `httpstatus`. Anyway, it's just an assumption, it's not the case in this moment so I'm going to change it. 👍